src/machine/machine_atmega328p.go

```diff
@@ -467,3 +472,93 @@ var SPI0 = SPI{
 	sdo:  PB3,
 	sdi:  PB4,
 	cs:   PB2}
+
+// Pin Change Interrupts
+var (
+	pinCallbacks [3][8]func(Pin)
+	interrupts   [3]*interrupt.Interrupt
+	previous     [3]uint8
+)
+
+func handlePCINTInterrupts(intr uint8, port *volatile.Register8) {
+	current := port.Get()
+	for i := uint8(0); i < 8; i++ {
+		if (current>>i)&0x01 == 0x01 && pinCallbacks[intr][i] != nil {
+			pinCallbacks[intr][i](Pin(i * (intr + 1)))
+		}
+	}
+	previous[intr] = current
+}
+
+func handlePCINT0Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT0-avr.IRQ_PCINT0, avr.PINB)
+}
+
+func handlePCINT1Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT1-avr.IRQ_PCINT0, avr.PINC)
+}
+
+func handlePCINT2Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT2-avr.IRQ_PCINT0, avr.PIND)
+}
+
+func SetInterrupt(pin Pin, callback func(Pin)) (err error) {
+	if callback == nil {
+		// Disable this pin interrupt
+		return ErrFeatureNotImplemented
+	}
+	var PCMSK_REG *volatile.Register8
+	var PCIE uint8
+	var PCINT uint8
+	var IRQ uint8
+	var PIN0 Pin
+
+	switch {
+	case pin >= PB0 && pin <= PB7:
+		// PCMSK0 - PCINT0-7
+		PCMSK_REG = avr.PCMSK0
+		PCIE = avr.PCICR_PCIE0
+		PIN0 = PD0
+		PCINT = 1 << (pin - PIN0)
+		IRQ = avr.IRQ_PCINT0
+		if interrupts[IRQ-avr.IRQ_PCINT0] == nil {
+			intr := interrupt.New(avr.IRQ_PCINT0, handlePCINT0Interrupts)
+			interrupts[IRQ-avr.IRQ_PCINT0] = &intr
+		}
```

Comment:
`interrupt.New` is a bit of a magical function and doesn't do anything at runtime (only at compile time). Therefore, I suggest using the following:

```suggestion
		interrupt.New(avr.IRQ_PCINT0, handlePCINT0Interrupts)
```

(same for the other 2 interrupts).

Reply:
Ah interesting...

Reply:
Hm, why did I think I needed to track all of the interrupts in an array?